src/complex/matrix.rs

```diff
@@ -0,0 +1,2290 @@
+        let l = r * c - 1;
+        let mut data: Vec<Complex<f64>> = self.data.clone();
+        let ref_data = &self.data;
+
+        match self.shape {
+            Shape::Row => {
+                for i in 0..l {
+                    let s = (i * c) % l;
+                    data[i] = ref_data[s];
+                }
+                data[l] = ref_data[l];
+                complex_matrix(data, r, c, Shape::Col)
+            }
+            Shape::Col => {
+                for i in 0..l {
+                    let s = (i * r) % l;
+                    data[i] = ref_data[s];
+                }
+                data[l] = ref_data[l];
+                complex_matrix(data, r, c, Shape::Row)
+            }
+        }
+    }
+
+    /// Change Bindings Mutably
+    ///
+    /// `Row` -> `Col` or `Col` -> `Row`
+    ///
+    /// # Examples
+    /// ```rust
+    /// use peroxide::fuga::*;
+    /// use num_complex::Complex64;
+    /// use peroxide::complex::matrix::*;
+    ///
+    /// let mut a = complex_matrix(vec![Complex64::new(1f64, 1f64),
+    ///                                 Complex64::new(2f64, 2f64),
+    ///                                 Complex64::new(3f64, 3f64),
+    ///                                 Complex64::new(4f64, 4f64)],
+    ///                             2, 2, Row
+    ///     );
+    /// assert_eq!(a.shape, Row);
+    /// let b = a.change_shape_mut();
+    /// assert_eq!(b.shape, Col);
+    /// ```
+    pub fn change_shape_mut(&mut self) -> Self {
+        let r = self.row;
+        let c = self.col;
+        assert_eq!(r * c, self.data.len());
+        let l = r * c - 1;
+        let mut data: Vec<Complex<f64>> = self.data.clone();
+        let ref_data = &self.data;
+
+        match self.shape {
+            Shape::Row => {
+                for i in 0..l {
+                    let s = (i * c) % l;
+                    data[i] = ref_data[s];
+                }
+                data[l] = ref_data[l];
+                complex_matrix(data, r, c, Shape::Col)
+            }
+            Shape::Col => {
+                for i in 0..l {
+                    let s = (i * r) % l;
+                    data[i] = ref_data[s];
+                }
+                data[l] = ref_data[l];
+                complex_matrix(data, r, c, Shape::Row)
+            }
+        }
+    }
+
+    /// Spread data(1D vector) to 2D formatted String
+    ///
+    /// # Examples
+    /// ```rust
+    /// use peroxide::fuga::*;
+    /// use num_complex::Complex64;
+    /// use peroxide::complex::matrix::*;
+    ///
+    /// let a = complex_matrix(vec![Complex64::new(1f64, 1f64),
+    ///                                 Complex64::new(2f64, 2f64),
+    ///                                 Complex64::new(3f64, 3f64),
+    ///                                 Complex64::new(4f64, 4f64)],
+    ///                             2, 2, Row
+    ///     );
+    /// println!("{}", a.spread()); // same as println!("{}", a);
+    /// // Result:
```

Comment:
Result shows still f64 matrix

Reply:
Done in PR: https://github.com/Axect/Peroxide/pull/71/commits/f2778957a20f24e9ef709482a4a7e0013c1b7788